rfcs/proposed/iterators_pass_directly_customization/README.md

```diff
@@ -0,0 +1,151 @@
+# Passed Directly Customization Point for User Defined Types
+
+## Introduction
+
+oneDPL handles some types of input data automatically as input to its dpcpp (sycl-based) backend as described
+[here](https://uxlfoundation.github.io/oneDPL/parallel_api/pass_data_algorithms.html). Unified Shared Memory (USM)
+pointers refer to data which is device accessible inherently, so no processing is required to pass this type of input
+data to SYCL kernels, we refer to this trait as "passed directly". oneDPL also defines some rules for its provided
+[iterator types](https://uxlfoundation.github.io/oneDPL/parallel_api/iterators.html) to be passed directly to SYCL
+under some circumstances (based usually on their base types).
+
+Internally, these rules are defined with a trait `oneapi::dpl::__ranges::is_passed_directly<T>` which evaluates to
+`std::true_type` or `std::false_type` to indicate whether the iterator type `T` should be passed directly to sycl
+kernels. There exists an unofficial legacy `is_passed_directly` trait which types can define like this:
+`using is_passed_directly = std::true_type;` which is supported within oneDPL. This method is currently used for a
+number of helper types within the SYCLomatic compatibility headers, (`device_pointer`, `device_iterator`,
+`tagged_pointer`, `constant_iterator`, `iterator_adaptor`). There is no official public API for users who want to
+create their own iterator types which could be passed directly to SYCL kernels, this is a gap that should be filled 
+with an official public API.
+
+Without something like this users are forced to only rely upon our provided iterator types, or reach into implementation
+details which are not part of oneDPL's specified interface.
+
+## Proposal
+
+Create a customization point `oneapi::dpl::is_passed_directly_in_onedpl_device_policies` free function which allows
+users to mark their types as passed directly:
+
+```
+template <typename T>
+constexpr bool is_passed_directly_in_onedpl_device_policies(const T&);
+```
+
+oneDPL will provide a default implementation which will defer to the existing trait:
```

Comment:
What header should a user include to get the default implementation? Execution? I think it would be good to specify it here.

Reply:
Its a good question, and I think I agree that execution is the appropriate place.

Reply:
`<oneapi/dpl/traits>`?

Reply:
Now that the RFC has evolved a bit to make the public API a trait rather than directly calling the customization point, I think `oneapi/dpl/type_traits` does make more sense than `oneapi/dpl/execution`.

Reply:
`oneapi/dpl/type_traits` makes sense to me as well.

Reply:
I think it should better go to `<oneapi/dpl/iterator>`.

`type_traits` is OK in principle, but, as far as I can tell, this thing is closer semantically to a concept representing an iterator category (like `forward_iterator`) or a requirement (like `sortable`) than to a generic type trait.

Reply:
I don't have a strong opinion between these two options.  You are correct that it specifically will augment a custom iterator type, so I don't object to `oneapi/dpl/iterator`.  The inclusion there may perhaps also help users find some combination of our provided custom iterators which may serve their purposes without requiring their own customizations in the first place.

Reply:
I've added an open question about where this should live.  We can discuss in the soon to come specification PR.